sky/server/common.py

```diff
@@ -217,9 +225,28 @@ def get_api_server_status(endpoint: Optional[str] = None) -> ApiServerInfo:
                         server_info.status = ApiServerStatus.VERSION_MISMATCH
                     return server_info
                 except (json.JSONDecodeError, AttributeError) as e:
+                    # Try to check if we got redirected to a login page.
+                    response = requests.get(f'{server_url}/api/login',
+                                            timeout=2.5,
+                                            cookies=get_api_cookie_jar(),
+                                            allow_redirects=False)
+                    logger.debug('Second attempt to check redirect: '
+                                 f'{response.status_code}')
+                    if response.status_code == 302:
+                        location = response.headers.get('Location', '')
+                        logger.debug(f'Redirect URL: {location}')
```

Comment:
What if there are multiple redirections? I mean, if the `server_url` does no have redirect configured at any of the proxys, then we can set `allow_redirects=False` in `/api/health` request and check `status_code==302` at the first place. If the `server_url` will be redirected to another host first, then we cannot detect whether an auth is required here, right? Or maybe I misunderstood the whole process...

Reply:
note to self: https://requests.readthedocs.io/en/latest/user/quickstart/#redirection-and-history